jerry-core/parser/js/js-lexer.c

```diff
@@ -448,6 +448,45 @@ static const keyword_string_t * const keyword_string_list[9] =
 #undef LEXER_KEYWORD
 #undef LEXER_KEYWORD_END
 
+#ifndef CONFIG_DISABLE_ES2015_SHORTHAND_OBJ_LITERAL_NOTATION
+/**
+ * Checks if current identifier is a keyword or reserved word.
+ *
+ * @return true - if the current identifier is a keyword or reserved word
+ *         false - otherwise
+ */
+bool
+lexer_is_identifier_keyword (parser_context_t *context_p) /**< context */
```

Comment:
There is a similar code for this. It would be good to not duplicate that.

You need to be smart here, invent a way which can be effectively optimized by tail merging or just "reparse" the string or whatever.

Reply:
I still need to figure out what to do about this. My updated patch doesn't yet address this.